WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -4075,8 +4076,19 @@ end
 
 function WeakAuras.SetDragging(data, drop)
   WeakAuras_DropDownMenu:Hide()
-  for id, button in pairs(displayButtons) do
-    button:SetDragging(data, drop)
+  if (frame.pickedDisplay == tempGroup and #tempGroup.controlledChildren > 0) then
+    local children = {}
+    for index, childId in pairs(tempGroup.controlledChildren) do
+      local data = WeakAuras.GetData(childId);
+      children[data.id] = index
+    end
+    for id, button in pairs(displayButtons) do
+      button:SetDragging(data and children, drop, displayButtons);
```

Comment:
Here for each button you drag, you call button:SetDragging, which will then iterate over all displayButtons, essentially each call to button:SetDragging doing the same work.

Reply:
button:SetDragging iterate over all displayButtons only once `(displayButtons and data[self.data.id] == 1)`